### PR TITLE
isl-devel: update to 0.25

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

There were issues with building `isl` 0.25 on some systems, which led to reverting the port upgrade: https://github.com/macports/macports-ports/pull/15391
@kencu made a fix for those, however PR was not merged: https://github.com/macports/macports-ports/pull/15558

Since we already have `isl-devel` sitting useless, put 0.25 there as an option.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

macOS 10.5.8
Xcode 3.1.4

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by barracuda156** [comment]
@kencu This should not hurt anyone, I guess? Your patch is included.

**Comment by barracuda156** [comment]
@michaelld Maybe you could help merging this? `-devel` version, optional, so cannot break anything for anyone. Works perfectly fine at least on a number of systems.

File: devel/isl-devel/Portfile [hunk 1]
```diff
@@ -1,15 +1,112 @@
...
+master_sites        sourceforge:libisl
+distname            ${my_name}-${version}
+use_bzip2           yes
+
+checksums           rmd160  73b9005b8e7fe52eb9500b98ba65450bdfb9f7fb \
+                    sha256  4305c54d4eebc4bf3ce365af85f04984ef5aa97a52e01128445e26da5b1f467a \
+                    size    2304378
+
+if {${subport} eq ${name}} {
+    platform darwin {
```
Comment (michaelld) [inline devel/isl-devel/Portfile:48]: does this need to be `platform darwin`? if not remove this clause

File: devel/isl-devel/Portfile [hunk 2]
```diff
@@ -1,15 +1,112 @@
...
+
+triplet.add_build   cross
+
+test.run            yes
+test.target         check
+
+livecheck.type      regex
+livecheck.regex     ${name}-(\[0-9.\]+)${extract.suffix}
+
+subport isl14 {
```
Comment (michaelld) [inline devel/isl-devel/Portfile:73]: remove this subport, as it is already part of `isl` Portfile

File: devel/isl-devel/Portfile [hunk 3]
```diff
@@ -1,15 +1,112 @@
...
+
+    # Don't link using "-flat_namespace -undefined suppress" on Yosemite and later.
+    patchfiles-append   yosemite-libtool.patch
+
+    configure.pre_args  --prefix=${prefix}/libexec/${subport}
+
+    livecheck.regex     isl-([join [lrange [split ${version} .] 0 2] .]\[0-9.\]*)${extract.suffix}
+}
+
+subport isl18 {
```
Comment (michaelld) [inline devel/isl-devel/Portfile:92]: remove this subport, as it is already part of `isl` Portfile

**Review by michaelld — CHANGES_REQUESTED** [review]
good start ... some changes required!

File: devel/isl/Portfile [hunk 4]
```diff
@@ -14,7 +15,6 @@ version             0.24
 revision            1
 
 categories          devel math
-platforms           darwin
```
[the remark this answers is not in the document]
Comment (barracuda156) [reply devel/isl/Portfile:17]: @michaelld From what I was told by @mascguy and others, it is the default, so redundant. AFAIU, only alternative cases should be specified (say, freebsd).
I in no way insist on removing this though – it is non-consequential.

**Comment by barracuda156** [comment]
@michaelld Thank you! I will make changes now.

File: devel/isl-devel/Portfile (hunk 1 above)
Comment (barracuda156) [reply devel/isl-devel/Portfile:48]: Yes, this is needed. All stuff inside the clause is Mac-specific.

**Comment by barracuda156** [comment]
@michaelld Good to merge now?

Re `darwin`, I have re-checked, it is the default, so not needed: “A list of the platforms on which the port is expected to work. Defaults to darwin if not set”.
https://guide.macports.org/#reference.keywords

**Comment by barracuda156** [comment]
@michaelld Could we merge this now?

File: devel/isl-devel/Portfile [hunk 5]
```diff
@@ -1,15 +1,71 @@
...
+
+master_sites        sourceforge:libisl
+distname            ${my_name}-${version}
+use_bzip2           yes
+
+checksums           rmd160  73b9005b8e7fe52eb9500b98ba65450bdfb9f7fb \
+                    sha256  4305c54d4eebc4bf3ce365af85f04984ef5aa97a52e01128445e26da5b1f467a \
+                    size    2304378
+
+if {${subport} eq ${name}} {
```
Comment (michaelld) [inline devel/isl-devel/Portfile:47]: If there are no subports except the main port, then this `if` isn't necessary. Thinking this is the case.

File: devel/isl-devel/Portfile (hunk 1 above)
Comment (michaelld) [reply devel/isl-devel/Portfile:48]: Gotcha. OK makes sense :+1:

File: devel/isl/Portfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (michaelld) [reply devel/isl/Portfile:17]: OK let's leave it as is, removing the line. Someone will want to fix it in the future if the port isn't Darwin-specific

File: devel/isl-devel/Portfile (hunk 5 above)
Comment (barracuda156) [reply devel/isl-devel/Portfile]: @michaelld Done!

**Comment by michaelld** [comment]
@barracuda156 almost there ... please convert tabs to spaces

**Comment by barracuda156** [comment]
> @barracuda156 almost there ... please convert tabs to spaces

@michaelld Oops, thanks for noticing. Fixed.

**Review by michaelld — APPROVED** [review]
LGTM ; thx!

**Comment by michaelld** [comment]
As this is a `devel` port, I'm good with merging it as-is & if there are any real issues we can clean them up down the road

**Comment by michaelld** [comment]
Thanks for your hard work and persistence @barracuda156  !

**Pull request merged**